concept/type_/type_reader.rs

```diff
@@ -147,43 +229,119 @@ impl TypeReader {
             .map_err(|error| ConceptReadError::SnapshotIterate { source: error })
     }
 
+    pub(crate) fn get_plays_transitive<T>(
+        snapshot: &impl ReadableSnapshot,
+        player: T,
+    ) -> Result<HashMap<RoleType<'static>, Plays<'static>>, ConceptReadError>
+    where
+        T: PlayerAPI<'static> + ReadableType<ReadOutput<'static> = T>, // ReadOutput=T is needed for supertype transitivity
+    {
+        // TODO: Should the player of a transitive plays be the declaring player or the inheriting player?
+        let mut transitive_plays: HashMap<RoleType<'static>, Plays<'static>> = HashMap::new();
+        let mut overridden_plays: HashSet<RoleType<'static>> = HashSet::new(); // TODO: Should this store the plays? This feels more fool-proof if it's correct.
+        let mut current_type = Some(player);
+        while current_type.is_some() {
+            let declared_plays = Self::get_plays(snapshot, current_type.as_ref().unwrap().clone())?;
+            for plays in declared_plays.into_iter() {
+                let role = plays.role();
+                if !overridden_plays.contains(&role) {
+                    debug_assert!(!transitive_plays.contains_key(&role));
+                    transitive_plays.insert(plays.role(), plays.clone());
+                    if let Some(overridden) = Self::get_plays_override(snapshot, plays.clone())? {
+                        overridden_plays.add(overridden.role());
+                    }
+                }
+            }
+            current_type = Self::get_supertype(snapshot, current_type.unwrap())?;
+        }
+        Ok(transitive_plays)
+    }
+
+    pub(crate) fn get_plays_override(
+        snapshot: &impl ReadableSnapshot,
+        plays: Plays<'static>,
+    ) -> Result<Option<Plays<'static>>, ConceptReadError> {
+        let override_property_key = build_property_type_edge_override(plays.into_type_edge());
+        snapshot
+            .get_mapped(override_property_key.into_storage_key().as_reference(), |overridden_edge_bytes| {
+                let overridden_edge = new_edge_plays(Bytes::Reference(overridden_edge_bytes));
```

Comment:
Should this be replaced with `IntoCanonicalTypeEdge` ?
We may even be able to make the override functions generic by making Plays & Owns readable types (!!!)